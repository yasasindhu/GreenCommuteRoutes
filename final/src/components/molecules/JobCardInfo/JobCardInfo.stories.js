import React from "react";

import JobCardInfo from "./JobCardInfo";

export default {
    title: "molecules/Job Card Info",
    component: JobCardInfo,
};
const Template = (args) => <JobCardInfo {...args} />;

export const PrimaryJobCardInfo = Template.bind({});


PrimaryJobCardInfo.args = {
    imageSrc : 'https://logosvector.net/wp-content/uploads/2016/05/hp-logo-vector-download.jpg',
    positionAppliedName : "User Experience Designer",
    companyName : "BMW",
    locationName : "Hitech City Telangana"
};

//bmw https://upload.wikimedia.org/wikipedia/commons/thumb/4/44/BMW.svg/1200px-BMW.svg.png
//microchip https://www.logolynx.com/images/logolynx/s_06/06968148ae0a9bd8f0990b0e6e332130.png
