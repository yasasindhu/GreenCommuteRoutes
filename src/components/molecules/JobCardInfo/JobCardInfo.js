import React from "react";
import { makeStyles } from '@material-ui/core/styles';
import MyTypography from "../../atoms/MyTypography/MyTypography";
import MyImage from '../../atoms/MyImage/MyImage';
import customtheme from '../../theme.jsx';
import { ThemeProvider } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import { Grid } from "@material-ui/core";

const JobCardInfo = ({ positionAppliedName, companyName, locationName, imageSrc }) => {

    const useStyles = makeStyles((theme) => ({
        root: {
            display: 'inline-flex',
            width: '390px',
            flexDirection: 'row',
            marginLeft: '30px',
        },
        cardDetailsDisplay: {
            display: 'flex',
            flexDirection: 'column',
            justifyContent: 'center',
            padding: '0 3% 3% 5%',
            paddingTop: '0',
            gap: '4px'
        },
        typographyHeaderStyle: {
            fontFamily: 'Montserrat',
            color: '#19293b',
            fontSize: '16px',
            lineHeight: 1.5,
            letterSpacing: '0.1px'
        },
        typographyParagraphStyle: {
            fontFamily: 'Montserrat',
            color: '#5f7381',
            fontSize: '12px',
            lineHeight: 1.33,
            letterSpacing: '0.1px'
        }

    }));

    const classes = useStyles();

    return (
        <Grid className={classes.root} data-testid="maindivinjobinfo">
            <ThemeProvider theme={customtheme}>
                <div data-testid="firstdivinjobinfo">
                    <MyImage src={imageSrc} height="50px" width="50px" alt={companyName} />
                </div>
                <div className={classes.cardDetailsDisplay} data-testid="seconddivinjobinfo">
                    <Typography variant="h2" data-testid="positiontypographyinjobinfo">{positionAppliedName}</Typography>
                    <Typography variant="h4" data-testid="companytypographyinjobinfo">{companyName}</Typography>
                    <Typography variant="h4" data-testid="locationtypographyinjobinfo">{locationName}</Typography>
                </div>
            </ThemeProvider>
        </Grid>
    );
}

export default JobCardInfo;