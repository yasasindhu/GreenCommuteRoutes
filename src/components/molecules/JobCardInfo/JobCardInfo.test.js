import { render, screen, cleanup } from '@testing-library/react';
import JobCardInfo from './JobCardInfo';





describe('Typography Rendered', () => {
    test('cardinjobcard', () => {
      const { asFragment} = render(<JobCardInfo positionAppliedName="Manager" companyName = "Zemoso" locationName = "Hyd" imageSrc='https://logosvector.net/wp-content/uploads/2016/05/hp-logo-vector-download.jpg'/>);
      expect(asFragment()).toMatchSnapshot();
    });
    
  });