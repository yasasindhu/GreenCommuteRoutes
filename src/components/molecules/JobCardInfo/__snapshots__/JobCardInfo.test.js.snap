// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography Rendered cardinjobcard 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root makeStyles-root-1"
    data-testid="maindivinjobinfo"
  >
    <div
      data-testid="firstdivinjobinfo"
    >
      <img
        alt="Zemoso"
        data-testid="image"
        height="50px"
        src="https://logosvector.net/wp-content/uploads/2016/05/hp-logo-vector-download.jpg"
        width="50px"
      />
    </div>
    <div
      class="makeStyles-cardDetailsDisplay-2"
      data-testid="seconddivinjobinfo"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2"
        data-testid="positiontypographyinjobinfo"
      >
        Manager
      </h2>
      <h4
        class="MuiTypography-root MuiTypography-h4"
        data-testid="companytypographyinjobinfo"
      >
        Zemoso
      </h4>
      <h4
        class="MuiTypography-root MuiTypography-h4"
        data-testid="locationtypographyinjobinfo"
      >
        Hyd
      </h4>
    </div>
  </div>
</DocumentFragment>
`;
